hydra-node/src/Hydra/Chain.hs

```diff
@@ -79,12 +81,12 @@ instance HasTypeProxy HeadId where
   proxyToAsType _ = AsHeadId
 
 instance Arbitrary HeadId where
-  arbitrary = genericArbitrary
+  arbitrary = HeadId . BS.pack <$> vectorOf 16 arbitrary
 
 -- | Describes transactions as seen on chain. Holds as minimal information as
 -- possible to simplify observing the chain.
 data OnChainTx tx
-  = OnInitTx {contestationPeriod :: ContestationPeriod, parties :: [Party]}
+  = OnInitTx {initHeadId :: HeadId, contestationPeriod :: ContestationPeriod, parties :: [Party]}
```

Comment:
The other fields are also not prefixed.

Should: name this `headId` (also in the schema)